qualtran/bloqs/mod_arithmetic/_shims.py

```diff
@@ -0,0 +1,250 @@
+#  Copyright 2023 Google LLC
+#
```

Comment:
Can you add a module level docstring explaining the need for _shims?

Reply:
added